typescript/infra/config/environments/mainnet2/agent.ts

```diff
@@ -59,10 +60,33 @@ const gasPaymentEnforcement: GasPaymentEnforcementConfig[] = [
     // all messages between interchain query routers.
     // This whitelist will become more strict with
     // https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1605
-    matchingList: interchainQueriesMatchingList,
+    matchingList: [
+      ...interchainQueriesMatchingList,
+      {
+        originDomain: [getDomainId(chainMetadata.solana)],
+        senderAddress: '*',
+        destinationDomain: '*',
+        recipientAddress: '*',
+      },
+      {
+        originDomain: [getDomainId(chainMetadata.bsc)],
+        senderAddress: ['0xC27980812E2E66491FD457D488509b7E04144b98'],
+        destinationDomain: '*',
+        recipientAddress: '*',
+      },
+    ],
   },
   {
     type: GasPaymentEnforcementPolicyType.OnChainFeeQuoting,
+    matchingList: [
+      {
+        originDomain: [getDomainId(chainMetadata.solana)],
+        // solana -> nautilus warp route
+        senderAddress: ['EJqwFjvVJSAxH8Ur2PYuMfdvoJeutjmH6GkoEFQ4MdSa'],
```

Comment:
I added this warp route under `OnChainFeeQuoting` since it now has an IGP configured